---
title: "Tabita_28_morning"
output: html_document
date: "2025-06-28"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

*************************************************************************************************************

Practice: EHH
I. Install the rehh R package
II. Load rehh R package
III. Use the following files
Chr2_EDAR_LWK_500K.recode.vcf #(African population)
Chr2_EDAR_CHS_500K.recode.vcf # (East Asian population)
IV. What is the profile of ancestral and derived haplotypes of the rs3827760 SNP in AFR and EAS?

```{r}
library(rehh)
```

1. Convert the data to haplohh format
Use the 'data2haplohh' function

```{r}
EDARlwk <- data2haplohh("C:\\Users\\Erica\\Desktop\\EMBO2025\\Tabita_Hunemeier\\EMBO_Practical_Course_2024-main\\Chr2_EDAR_LWK_500K.recode.vcf", polarize_vcf = FALSE)
EDARchs <- data2haplohh("C:\\Users\\Erica\\Desktop\\EMBO2025\\Tabita_Hunemeier\\EMBO_Practical_Course_2024-main\\Chr2_EDAR_CHS_500K.recode.vcf", polarize_vcf = FALSE)
```

2. Calculate the EHH for the candidate SNP (rs3827760) in AFR
Use the 'calc_ehh' function
3. Calculate the EHH for the candidate SNP (rs3827760) in EAS

```{r}
AFR_ehh <- calc_ehh(EDARlwk, mrk="rs3827760", polarized = FALSE)
EAS_ehh <- calc_ehh(EDARchs, mrk="rs3827760", polarized = FALSE)
```

4. Plot EHH around “rs3827760” in AFR and EAS

```{r}
par(mfrow = c(1, 2))
plot(AFR_ehh, main = "LWK")
plot(EAS_ehh, main = "CHS")
dev.off()
```

6. Calculate furcation trees around a candidate SNP in AFR
7. Calculate furcation trees around a candidate SNP in EAS

```{r}
par(mfrow = c(1, 2))
plot(calc_furcation(EDARlwk,  mrk="rs3827760"), main="LWK")
plot(calc_furcation(EDARchs,  mrk="rs3827760"), main="CHS")
dev.off()
```


*********************************************************************************************

Practice 2: integrated Haplotype score (iHS)

1. Calculate the EHH for all SNPs in the file for AFR
Use the 'scan_hh' function
2. Calculate the EHH for all SNPs in the file for EAS

```{r}
EDARlwk_hh <- scan_hh(EDARlwk, polarized = FALSE)
EDARchs_hh <- scan_hh(EDARchs, polarized = FALSE)
```






